File: 프로그래머스/lv1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.js
function solution(numbers) {
    let answer = [];
    for(let i = 0; i < numbers.length; i++){
        for(let j = i + 1; j < numbers.length; j++){
            answer.push(numbers[i] + numbers[j]);
        }
    }
    answer.sort((a,b) => {
        if(a > b) return 1;
        else if(a < b) return -1;
        else return 0;
    });
    return answer.filter((element, index) => {
        return answer.indexOf(element) === index;
    });
}

File: 프로그래머스/lv1/68644. 두 개 뽑아서 더하기/두 개 뽑아서 더하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] numbers) {
        
        PriorityQueue<String> queue = new PriorityQueue<>();
        
        for(int i = 0; i < numbers.length; i++){
            for(int j = i + 1; j < numbers.length; j++){
                queue.offer(Integer.toString(numbers[i] + numbers[j]));
            }
        }
        String[] answer = new String[queue.size()];
        int cnt = 0;
        while(queue.size() > 0){
            if(answer[0] == null){
                answer[cnt] = queue.poll();
                cnt++;
            }else{
                if(queue.peek() == answer[cnt - 1]) queue.poll();
                else{
                    answer[cnt] = queue.poll();
                    cnt++;
                }
            }
        }
        
        return Arrays.stream(answer).mapToInt(Integer::parseInt).distinct().sorted().toArray();
    }
}